src/main/java/Model/Game.java

```diff
@@ -1,269 +1,383 @@
      */
     private void disableEnPassant() {
         for (Piece piece:enPassantAvailablePieces) {
             piece.setEnPassantAvailableToTakeFile(null);
         }
-        this.enPassantPosition = null;
+        this.enPassantPositionForFen = null;
         enPassantAvailablePieces.clear();
     }
 
+
+    /**
+     * Update any pawn that can en Passant next turn and then add those pieces that
+     * are able to en passant on the next turn to the enPassantAvailablePieces list
+     * 
+     * @param newPosition new position of the pawn
+     * @param isWhite if the pawn is white or black
+     */
+    
+    private void enableEnPassant(Position newPosition, boolean isWhite) {
+        int tempFile;
+        for (int iFile = -1; iFile < 2; iFile+=2) {
+            tempFile = newPosition.getFile() + iFile;
+            if (tempFile >= 0 && tempFile < 8){
+                Piece piece = board.getPiece(new Position(tempFile,newPosition.getRank()));
+                if(nonNull(piece) && piece.getPieceType() == PieceType.Pawn && piece.getIsWhite() != isWhite){
+                    piece.setEnPassantAvailableToTakeFile(newPosition.getFile());
+                    enPassantAvailablePieces.add(piece);
+                }
+            }
+        }
+    }
+
+
     /**
+     * WIP!
      * Check to see if a pawn has made it to the other side of the board. If so call the promotion function
      * 
-     * @param newRank
-     * @param piece
+     * @param newPosition of the piece
+     * @param piece that just moved
      */
-    private void checkForPromotion(int newRank, Piece piece) {
+    private void checkForPromotion(Position newPosition, Piece piece) {
         if(piece.getPieceType() != PieceType.Pawn) {
             return;
         }
         int rank = piece.getIsWhite() ? 0 : 7;
-        if(newRank != rank){
+        if(newPosition.getRank() != rank){
             return;
         }
         piece.pawnPromotion(PieceType.Queen);
     }
 
 
     /**
-     * Update the FEN and then update the view
-     * 
-     * @param newPosition
-     * @param previousPosition
+     * Call updateView function in the observer
+     *
      */
-    public void updateView(Position newPosition, Position previousPosition) {
+    private void updateView() {
         _observer.update();
     }
 
 
     /**
-     * Call chooseMove in Moves class
+     * Change whose turn it is
+     * update the fullMove if required
+     * check to see if it's a checkmate or stalemate
+     */
+    private void endTurn() {
+        whitesTurn = !whitesTurn;
+        fullMove +=whitesTurn?1:0;
+
+        gameOver();
+    }
+
+    private void gameOver(){
+        if (Check.isThereALegalMove(this, whitesTurn)){
+            if (Check.isTheKingUnderAttack(this, whitesTurn)){
+                System.out.println("CHECKMATE BITCHES!!");
+            }
+            else {
+                System.out.println("STALEMATE??");
+            }
+        }
+    }
+
+
+
+
+
+
+
+
+
+
+
+
+
```

Comment:
wtf is this

Reply:
So Game is huge, but can be broken down into sections.

- initialization
- making a move
- get possible moves
- getters

I find it easier to navigate the file when I have these big breaks inbetween them all. Will get rid of these spaces when I'm finished though